lib/ldap_fluff/ad_member_service.rb

```diff
@@ -32,12 +32,15 @@ def _walk_group_ancestry(group_dns = [])
     group_dns.each do |group_dn|
       search = @ldap.search(:base => group_dn, :scope => Net::LDAP::SearchScope_BaseObject, :attributes => ['memberof'])
       if !search.nil? && !search.first.nil?
-        group = search.first
-        set  += _walk_group_ancestry(group[:memberof])
-        set  += group[:memberof]
+        groups = search.first[:memberof] - known_groups
```

Comment:
Could this just use the existing list of gruops?  The arrays look like they'll probably contain the same data - they're adding both `groups` and `next_level` on every iteration.